ldap/servers/slapd/main.c

```diff
@@ -499,15 +499,22 @@ write_start_pid_file(void)
      * admin programs. Please do not make changes here without
      * consulting the start/stop code for the admin code.
      */
-    if ((start_pid_file != NULL) && (fp = fopen(start_pid_file, "w")) != NULL) {
+    if (start_pid_file == NULL) {
+        return 0;
+    } else if (fp = fopen(start_pid_file, "w") != NULL) {
```

Comment:
line 504 is wrong as the following warning shows. 
Should be ((fp = fopen(...)) != ... )

Reply:
That's an easy fix :)